src/vsr/replica.zig

```diff
@@ -1145,66 +1276,302 @@ pub fn Replica(
+                for (self.recovery_response_from_other_replicas) |response, replica| {
+                    if (response == null) continue;
+                    assert(replica != self.replica);
+                    assert(response.?.header.replica == replica);
+                    assert(response.?.header.context == self.recovery_nonce);
+
+                    view = std.math.max(view, response.?.header.view);
+                }
+                break :blk view;
+            };
+
+            const leader_replica = self.leader_index(view);
+            const leader_response = responses[leader_replica];
+            if (leader_response == null) {
+                log.debug("{}: on_recovery_response: ignoring (awaiting response from leader of view={})", .{
+                    self.replica,
+                    view,
+                });
+                return;
+            }
+
+            if (leader_response.?.header.view != view) {
+                // The leader (according to the view quorum) isn't the leader (according to itself).
+                // The `recovery_timeout` will retry shortly with another round.
+                log.debug("{}: on_recovery_response: ignoring (leader has wrong view actual={} expect={})", .{
+                    self.replica,
+                    leader_response.?.header.view,
+                    view,
+                });
+                return;
+            }
+
+            // This recovering→normal status transition occurs exactly once.
+            // All further `recovery_response` messages are ignored.
+
+            // TODO When the view number is recovered from the superblock (instead of VSR recovery
+            //   protocol), if the view number indicates that this replica is a leader, it must
+            //   transition to status=view_change instead of status=normal.
+
+            const leader_headers = self.message_body_as_headers(leader_response.?);
+            assert(leader_headers.len > 0);
+
+            const commit = leader_response.?.header.commit;
+            {
+                var latest = Header.reserved(self.cluster, 0);
+                set_latest_op(leader_headers, &latest);
+                assert(latest.op == leader_response.?.header.op);
+
+                self.set_latest_op_and_k(&latest, commit, "on_recovery_response");
+                assert(self.journal.header_with_op(self.op) != null);
+            }
+
+            assert(self.status == .recovering);
+            self.transition_from_recovering_status(view);
+            assert(self.status == .normal);
+            assert(self.follower());
+
+            // TODO if the view's primary is >1 WAL ahead of us, these headers could cause problems. We don't want to jump this far ahead to repair, but we still need to use the hash chain to figure out which headers to request. Maybe include our op_checkpoint in the recovery (request) message so that the response can give more useful (i.e. older) headers.
```

Comment:
Let's wrap this to a hard limit of 100 columns. There may be a few other places where we need to wrap at 100 columns. Running `zig build lint` will also help point these out.